pkg/api/v20170701/validate.go

```diff
@@ -157,9 +161,18 @@ func (a *Properties) Validate() error {
 	if e := validateUniqueProfileNames(a.AgentPoolProfiles); e != nil {
 		return e
 	}
+
 	if a.OrchestratorProfile.OrchestratorType == Kubernetes {
-		if a.ServicePrincipalProfile == nil {
```

Comment:
How does this tie in with @colemickens change where service principal id & secret can be auto generated?